### DISCUSSION TITLE
[QUESTION] PROTEST implementation

### DISCUSSION THREAD

**Discussion by valentynbez** [vegandevs/vegan#560] [Q&A] [answered]
In original Jackson 1995 the procedure for permutational testing is: 
- permute
- gather $M^2$ values
- select a fraction that is < original $M^2$
 
In the `vegan` implementation procedure goes as follows: 
- permute 
- gather $M^2$ values
- select a fraction that is >= original $\sqrt{1 - M^2}$

https://github.com/vegandevs/vegan/blob/1c98b986903666649548a0a1f01595a024fd84dc/R/protest.R#L19-L38

What is the reason for choosing such an approach? (I am not the strongest in maths, it might be that I'm missing simple logic somewhere).

Also, as far as I understand, $m_{12}^2$ refers to the sum of squares or goodness-of-fit statistic from the publication of Peres-Neto & Jackson (2001). The same Jackson in 1995 refers to it as $M^2$. 
Do you think the notation could be unified to avoid confusion? 
I compared implementations in `QIIME2` and `vegan` both, you can take a look at it here: https://forum.qiime2.org/t/procrustes-and-protest-analysis/20123/4

**Comment by jarioksa** [comment, marked as answer]
`vegan::protest` gathers and compares Procrustes correlation _r_ instead of symmetric Procrustes sum of squares _m_<sup>2</sup>. These are inversely related as $r = \sqrt{1-m^2}$: when _m_<sup>2</sup> goes down, _r_ goes up and therefore the direction of inequality was changed in comparisons. However, same results will be obtained with both.

There is no special reason for this approach: there are several approaches that give the same results, and I just picked one of equal alternatives.

As to notation: **vegan** mentions $m_{12}^2$ in help page, but does not use that symbol in printed output or calculations. I don't know why Jackson opted for using uppercase _M_ in 1995 and lowercase _m_ in 2001 (with Peres-Neto), but I have no way to retroactively unify his notation. **Vegan** bases `protest` on the 2001 paper and therefore uses that notation. The Procrustes rotation (function `procrustes`) was based on Mardia _et al_ (1979) and the basic notation is taken from that book; `protest` was added later, and therefore it also basically uses the same notation as `procrustes`.

**Reply by valentynbez** [reply]
Thanks for the clarification! 
Another question I have is that `vegan` limits the number of available permutations by the number of samples. This helps to avoid identical permutations in both groups. 
Does this have any implications on a p-value? I assume if Nperm > Nobs p-value becomes unreliable, as we obtain repeated samples, is that right?

**Reply by jarioksa** [reply]
What do you mean with this? **Vegan** does not limit the number of permutations, except in the case when the user requests a higher number of permutations than the maximum number of possible permutations (and this limitation is made in the **permute** package, not in **vegan**). In that case **permute** switches to complete enumeration of all possible permutations and to so-called exact test.
